### ISSUE TITLE
Align with recent cfg80211 header

### ISSUE THREAD

**Issue by jychen0611** [sysprog21/vwifi#61]
### **1. The scan_width field is no longer present in the cfg80211_inform_bss structure.**

_[commit 5add321](https://github.com/torvalds/linux/commit/5add321c329b1746589b51359259666ca3dbe219) wifi: cfg80211: remove scan_width support_
kernel version : _v6.9-rc6 ~ v6.7-rc1_
date : Sep 13, 2023

> There really isn't any support for scanning at different channel widths than 20 MHz since there's no way to set it. Remove this support for now, if somebody wants to maintain this whole thing later we can revisit how it should work.

```diff
- bss_data.scan_width = NL80211_BSS_CHAN_WIDTH_20;
```

### **2. The parameters in the change_beacon function have been updated to cfg80211_ap_update.**
_[commit bb55441](https://github.com/torvalds/linux/commit/bb55441c57ccc5cc2eab44e1a97698b9d708871d) wifi: cfg80211: split struct cfg80211_ap_settings_
kernel version : _v6.9-rc6 ~ v6.7-rc1_
date : Sep 25, 2023

> Using the full struct cfg80211_ap_settings for an update is misleading, since most settings cannot be updated. Split the update case off into a new struct cfg80211_ap_update.
```diff
/**
 * struct cfg80211_ap_update - AP configuration update
 *
 * Subset of &struct cfg80211_ap_settings, for updating a running AP.
 *
 * @beacon: beacon data
 * @fils_discovery: FILS discovery transmission parameters
 * @unsol_bcast_probe_resp: Unsolicited broadcast probe response parameters
 */

+struct cfg80211_ap_update {
+	struct cfg80211_beacon_data beacon;
+	struct cfg80211_fils_discovery fils_discovery;
+	struct cfg80211_unsol_bcast_probe_resp unsol_bcast_probe_resp;
+};


	int	(*change_beacon)(struct wiphy *wiphy, struct net_device *dev,
-				 struct cfg80211_ap_settings *info);
+				 struct cfg80211_ap_update *info);
```
_[commit 66f85d5 ](https://github.com/torvalds/linux/commit/66f85d57b7109baf8a7d5ee04049ac9412611d35)wifi: cfg80211: modify prototype for change_beacon_
kernel version : _v6.9-rc6 ~ v6.7-rc1_
date : Sep 13, 2023

> Modify the prototype for change_beacon() in struct cfg80211_op to accept cfg80211_ap_settings instead of cfg80211_beacon_data so that it can process data in addition to beacons. Modify the prototypes of ieee80211_change_beacon() and driver specific functions accordingly.

```diff
int	(*change_beacon)(struct wiphy *wiphy, struct net_device *dev,
-				 struct cfg80211_beacon_data *info);
+				 struct cfg80211_ap_settings *info);
```
Simply put, the change_beacon function now includes two additional parameters, fils_discovery and unsol_bcast_probe_resp, which are part of the cfg80211_ap_update structure.

**_fils_discovery_** : FILS discovery transmission parameters
**_unsol_bcast_probe_resp_** : Unsolicited broadcast probe response parameters

**Comment by jserv** [comment]
Close via commit 35ca058e767d87a7c809091ae910b2bf02687870
